presto-jdbc/src/test/java/io/prestosql/jdbc/BaseTestJdbcResultSet.java

```diff
@@ -90,6 +91,84 @@ public void testDecimal()
+            checkRepresentation(connectedStatement.getStatement(), String.format("DECIMAL '%s'",
+                    outsideOfDoubleExactRange), Types.DECIMAL,
+                    (rs, column) -> {
+                        assertEquals(rs.getObject(column), new BigDecimal("9223372036854775774"));
+                        assertEquals(rs.getBigDecimal(column), new BigDecimal("9223372036854775774"));
+                        assertEquals(rs.getLong(column), 9223372036854775774L);
+                        assertEquals(rs.getDouble(column), 9.223372036854776E18);
+                        assertEquals(rs.getString(column), "9223372036854775774");
+                    });
+
+            checkRepresentation(connectedStatement.getStatement(), "VARCHAR ''", Types.VARCHAR, (rs, column) -> {
+                assertThatThrownBy(() -> rs.getBigDecimal(column))
+                        .isInstanceOf(SQLException.class)
+                        .hasMessage("Value is not a number: ");
+            });
+
+            checkRepresentation(connectedStatement.getStatement(), "VARCHAR '123a'", Types.VARCHAR, (rs, column) -> {
+                assertThatThrownBy(() -> rs.getBigDecimal(column))
+                        .isInstanceOf(SQLException.class)
+                        .hasMessage("Value is not a number: 123a");
+            });
+        }
+    }
+
+    @Test
+    public void testLong()
+            throws Exception
+    {
+        try (ConnectedStatement connectedStatement = newStatement()) {
+            checkRepresentation(connectedStatement.getStatement(), "VARCHAR '123'", Types.VARCHAR,
+                    (rs, column) -> assertEquals(rs.getLong(column), 123L));
+
+            checkRepresentation(connectedStatement.getStatement(), "VARCHAR ''", Types.VARCHAR, (rs, column) -> {
+                assertThatThrownBy(() -> rs.getLong(column))
+                        .isInstanceOf(SQLException.class)
+                        .hasMessage("Value is not a number: ");
+            });
+
+            checkRepresentation(connectedStatement.getStatement(), "VARCHAR '123a'", Types.VARCHAR, (rs, column) -> {
+                assertThatThrownBy(() -> rs.getLong(column))
+                        .isInstanceOf(SQLException.class)
+                        .hasMessage("Value is not a number: 123a");
+            });
+        }
+    }
+
+    @Test
+    public void testDouble()
+            throws Exception
+    {
+        try (ConnectedStatement connectedStatement = newStatement()) {
+            checkRepresentation(connectedStatement.getStatement(), "DOUBLE '123.456'", Types.DOUBLE,
+                    (rs, column) -> assertEquals(rs.getDouble(column), 123.456D));
+
+            checkRepresentation(connectedStatement.getStatement(), "VARCHAR '123'", Types.VARCHAR,
+                    (rs, column) -> assertEquals(rs.getDouble(column), 123D));
+
+            checkRepresentation(connectedStatement.getStatement(), "VARCHAR ''", Types.VARCHAR, (rs, column) -> {
```

Comment:
both doubles and varchars are tested in testPrimitiveTypes currently.
lets move it there for now

Reply:
Moved them inside the testPrimitiveTypes. I felt ```testPrimitiveTypes``` was kind of weird to put these tests, but it makes sense based on your comment earlier. I like the idea of breaking up the method ```testPrimitiveTypes``` into smaller specific methods in the future.